Feature: Add hide capability to buttons on sidebar (Settings > Appearance) and hamburger menu on sidebar for icons.

SpecialAro opened the pull request:
<!-- Thank you for your Pull Request. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!-- Please start by naming your pull request properly for e.g. "Add Google Tasks to Todo providers". -->
<!-- Please keep in mind that any text inside "<!--" and "--\>" are comments from us and won't be visible in your bug report, so please don't put any text in them. -->

#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has _stopped working_ or is _working incorrectly_, please log the bug [here](https://github.com/ferdium/ferdium-recipes/issues)
2. If you are requesting support for a **new service** in Ferdium, please log it [here](https://github.com/ferdium/ferdium-recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/ferdium/ferdium-app#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdium, but to get new recipes between Ferdium releases, this documentation is quite useful.)
4. Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change
Feature: Add hide capability to buttons on sidebar (Settings > Appearance).
Feature: Add hamburger menu on sidebar for icons.

#### Motivation and Context
Feature: Add hide capability to buttons on sidebar (Settings > Appearance), solving issue https://github.com/ferdium/ferdium-app/issues/65.
Feature: Add hamburger menu on sidebar for icons following https://github.com/ferdium/ferdium-app/issues/65 suggestions.

#### Screenshots
![image](https://user-images.githubusercontent.com/37463445/166169569-09471701-a1b0-45fa-9f11-063d9cb318c9.png)

#### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

#### Release Notes
Feature: Add hide capability to buttons on sidebar (Settings > Appearance)
Feature: Add hamburger menu on sidebar for icons.

SpecialAro committed:
Add hide feature to buttons on sidebar

vraravam commented:
I thought you were putting these within a hamburger menu?

SpecialAro commented:
You're right!
I'll implement it today so we can have both functionalities working 😁

SpecialAro committed:
Add Hambuger Menu

SpecialAro commented:
Added hamburger menu, as suggested
![image](https://user-images.githubusercontent.com/37463445/166231425-375edcb2-316d-4308-9ec4-9e22fdf73048.png)
![image](https://user-images.githubusercontent.com/37463445/166231441-85e8141d-3ee5-4ef8-81c8-61b61bf93346.png)

santhosh-chinnasamy commented on src/components/layout/Sidebar.js:
```diff
@@ -112,6 +115,10 @@ class Sidebar extends Component {
   }
 
+  collapseMenu() {
+    this.setState({ isCollapsed: !this.state.isCollapsed });
```
Hi @SpecialAro can we change this line to something like this
`this.setState(({ isCollapsed }) => ({ isCollapsed: !isCollapsed }));`

SpecialAro replied:
Will implement this on next commit! Thank you :D

MurzNN commented:
> Added hamburger menu, as suggested ![image](https://user-images.githubusercontent.com/37463445

Maybe a hamburger icon will be better than arrow here, to understand "what's this?" for users?

SpecialAro committed:
Update collapseMenu function

SpecialAro committed:
Final tweaks

SpecialAro commented:
> Maybe a hamburger icon will be better than arrow here, to understand "what's this?" for users?

Agree with you. Changed the icon and the default to not collapsed.

SpecialAro committed:
Trying to solve build error

vraravam commented:
Testing this now on my local before the approval! Great job @SpecialAro !

SpecialAro commented:
> Testing this now on my local before the approval! Great job @SpecialAro !

Thank you @vraravam ! And thank you everyone that helped me on this!

SpecialAro committed:
Fix horizontal bar icon

SpecialAro committed:
Add option to collapse settings button with Hamburger Menu

vraravam committed:
Renamed 'collapseSettingsButton' variable for consistency with other functionality from this same PR

vraravam committed:
Made 'hideSettings' follow the same pattern as the other 'hide*' buttons
Also reordered the 'Settings' button in sidebar closer to the other buttons being hidden

SpecialAro committed:
Hides hamburger menu if active buttons is less or equal than 1

vraravam commented:
LGTM! Great work @SpecialAro !
